low-level/framework/sspl_ll_d

```diff
@@ -613,6 +613,11 @@ def signal_handler(signal_number, frame):
        The entries in text file should be in form of <key=value>.
     """
     logger.debug("signal_handler called with {0}".format(signal_number))
+
+    # Ignore a new  SIGHUP while handling current SIGHUP
+    # Signal will be enabled at the end of this handler
+    signal.signal(signal.SIGHUP, signal.SIG_IGN)
```

Comment:
I would make this a first statement to be executed after we enter the signal handler. Was there any issue seen in that?

Reply:
Yes we can do that. 
FYI I never saw the log above i.e. logger.debug("signal_handler called with {0}".format(signal_number))  
coming up on VM in my testing ....
Its not visible in the quick experiment that was done either...
Any idea why that might be ?

Reply:
Did you check the logging level?

Reply:
Did not check ; but though that since 'info' is getting displayed; 'debug' should get displayed to; info is the lowest level rt ? 
Nyways will move the statement

Reply:
> /opt/seagate/cortx/hare/bin/consul kv put sspl/config/SYSTEM_INFORMATION/log_level DEBUG

@ArchanaLimaye change to DEBUG log level and you should see that log. And if this is working fine, then you should see this log only once.

@indrajitzagade agree, though barring logger.debug line its first line only. To check i think have log before signal is fine, or we would never know while testing or in production that signal handler indeed ignoring second signal, while first signal is still under processing.

Reply:
i think debug is the lowest level.

Reply:
Instead of this, can we use a global lock variable that will be set to True when handling the signal? If we change the signal handler itself, we may never know that a signal is received or not in the middle of handling another signal.

Reply:
@cdeshmukh I moved the SIG_IGN to before log message based on Indrajit's comment.....
should this be reverted back ? 

@thavanathan you mean some variable in consul or file ? 
>>we may never know that a signal is received or not in the middle of handling another signal.
Also do we need to be aware if the same signal is received again ?  SIG_IGN applies only to SIGHUP

Reply:
this handler is for catching SIGHUP, so that is enough for this one.